tools/annotation_tools/turk_with_s3/get_results.py

```diff
@@ -35,6 +43,92 @@ def get_hit_list_status(mturk):
+
+
+def get_hit_result(mturk, hit_id, output_csv: str, use_sandbox: bool, timeout: int):
+    if os.path.exists(output_csv):
+        res = pd.read_csv(output_csv)
+    else:
+        res = pd.DataFrame()
+
+    start_time = time.time()
+    status = ""
+    while status != "Reviewable":
+        if time.time() - start_time > timeout * 60:
+            logging.info(f"HIT {hit_id} is timeout")
+            delete_hit(mturk, hit_id)
+            return
+        status = get_hit_status(mturk, hit_id)
+        logging.info(f"HIT {hit_id} is not reviewable yet, currently is {status}...")
+        time.sleep(HIT_POLL_TIME)
+    
+    new_row = {"HITId": hit_id}
+    worker_results = mturk.list_assignments_for_hit(
+        HITId=hit_id, AssignmentStatuses=["Submitted"]
+    )
+    while worker_results["NumResults"] <= 0:
+        logging.info(f"HIT {hit_id} is reviewable, but not results can be retrieved yet...")
+        if time.time() - start_time > timeout * 60:
+            logging.info(f"HIT {hit_id} is timeout")
+            delete_hit(mturk, hit_id)
+            return
+        worker_results = mturk.list_assignments_for_hit(
+            HITId=hit_id, AssignmentStatuses=["Submitted"]
+        )
+        time.sleep(HIT_POLL_TIME)
+
+    if worker_results["NumResults"] > 0:
+        for assignment in worker_results["Assignments"]:
+            new_row["WorkerId"] = assignment["WorkerId"]
+            xml_doc = xmltodict.parse(assignment["Answer"])
+
+            logging.info("Worker's answer was:")
+            if type(xml_doc["QuestionFormAnswers"]["Answer"]) is list:
+                # Multiple fields in HIT layout
+                for answer_field in xml_doc["QuestionFormAnswers"]["Answer"]:
+                    input_field = answer_field["QuestionIdentifier"]
+                    answer = answer_field["FreeText"]
+                    logging.info("For input field: " + input_field)
+                    logging.info("Submitted answer: " + answer)
+                    new_row["Answer.{}".format(input_field)] = answer
+
+                res = res.append(new_row, ignore_index=True)
+                res.to_csv(output_csv, index=False)
+            else:
+                # One field found in HIT layout
+                answer = xml_doc["QuestionFormAnswers"]["Answer"]["FreeText"]
+                input_field = xml_doc["QuestionFormAnswers"]["Answer"]["QuestionIdentifier"]
+                logging.info("For input field: " + input_field)
+                logging.info("Submitted answer: " + answer)
+                new_row["Answer.{}".format(input_field)] = answer
+                res = res.append(new_row, ignore_index=True)
+                res.to_csv(output_csv, index=False)
+
+            mturk.approve_assignment(AssignmentId=assignment["AssignmentId"])
+            mturk.delete_hit(HITId=hit_id)
+    else:
+        logging.info("No results ready yet")
+        # if returned assignment is empty,reject
+        mturk.delete_hit(HITId=hit_id)
+    
```

Comment:
@kavyasrinet there're a lot of duplicate code due to the similarities between tool A/B/C/D. You should focus on this function.